File: benchmark/comparison/results/blind_py/29.py
class Ok:
    def __init__(self, v):
        self.v = v
    def __repr__(self):
        return f"Ok({self.v})"

class Err:
    def __init__(self, msg):
        self.msg = msg
    def __repr__(self):
        return f"Err({self.msg})"

def parse_int(s):
    try:
        return Ok(int(s))
    except ValueError:
        return Err(f"not a number: {s}")

def compute(a, b):
    if isinstance(a, Err):
        return a
    if isinstance(b, Err):
        return b
    return Ok(a.v + b.v)

def describe_result(r):
    if isinstance(r, Ok):
        return f"Success: {r.v}"
    return f"Error: {r.msg}"

print(parse_int("42"))
print(parse_int("abc"))
print(describe_result(compute(Ok(3), Ok(4))))
print(describe_result(compute(Err("bad"), Ok(4))))
print(describe_result(compute(Ok(3), Err("bad"))))
print(describe_result(compute(parse_int("42"), parse_int("10"))))
print(describe_result(compute(parse_int("42"), parse_int("abc"))))

File: benchmark/comparison/results/blind_ts/29.ts
type Result = { tag: "Ok"; value: number } | { tag: "Err"; msg: string };

function Ok(value: number): Result { return { tag: "Ok", value }; }
function Err(msg: string): Result { return { tag: "Err", msg }; }

function formatResult(r: Result): string {
  if (r.tag === "Ok") return `Ok(${r.value})`;
  return `Err(${r.msg})`;
}

function parse_int(s: string): Result {
  const n = Number(s);
  if (!isNaN(n) && s.trim() !== "" && Number.isInteger(n)) return Ok(n);
  return Err("not a number: " + s);
}

function compute(a: Result, b: Result): Result {
  if (a.tag === "Err") return a;
  if (b.tag === "Err") return b;
  return Ok(a.value + b.value);
}

function describe_result(r: Result): string {
  if (r.tag === "Ok") return `Success: ${r.value}`;
  return `Error: ${r.msg}`;
}

console.log(formatResult(parse_int("42")));
console.log(formatResult(parse_int("abc")));
console.log(describe_result(compute(Ok(3), Ok(4))));
console.log(describe_result(compute(Err("bad"), Ok(4))));
console.log(describe_result(compute(Ok(3), Err("bad"))));
console.log(describe_result(compute(parse_int("42"), parse_int("10"))));
console.log(describe_result(compute(parse_int("42"), parse_int("abc"))));
